Robot.java

```diff
@@ -0,0 +1,951 @@
+        }
+        else{
+          m_Intake.set(0);
+          m_Feeder.stopMotor();
+        }
+      }
+      if(intakeExtended){
+        s_LeftIntake.set(true);
+        s_RightIntake.set(true);
+        if(j_Operator.getRawButton(3)){
+          intakeExtended = false;
+        }
+      }
+      else{
+        s_LeftIntake.set(false);
+        s_RightIntake.set(false);
+      }
+    }    
+
+    public void visionTracking() {
+      if (chameleon_Yaw < -2) {
+        pc_Right1.setReference(-500, ControlType.kVelocity);
+        pc_Right2.setReference(-500, ControlType.kVelocity);
+        pc_Left1.setReference(-500, ControlType.kVelocity);
+        pc_Left2.setReference(-500, ControlType.kVelocity);
+
+      }
+
+      else if (chameleon_Yaw > 2) {
+        pc_Right1.setReference(500, ControlType.kVelocity);
+        pc_Right2.setReference(500, ControlType.kVelocity);
+        pc_Left1.setReference(500, ControlType.kVelocity);
+        pc_Left2.setReference(500, ControlType.kVelocity);
+      }
+
+      else {
+        m_Left1.stopMotor();
+        m_Left2.stopMotor();
+        m_Right1.stopMotor();
+        m_Right2.stopMotor();
+      }
+    }
+
+    public void intakingBalls() {
+      newBallBoolean = interruptSensor.get();
+      if(oldBallBoolean != newBallBoolean && newBallBoolean == true && ballDebounceBoolean == false){
+        //j_XboxController.setRumble(RumbleType.kLeftRumble, .5);
+        ballCounter++;
+        if(ballCounter <= 3) {
+          e_Feeder.setPosition(0);
+          pc_Feeder.setReference(120, ControlType.kPosition);
+        }
+        else if (ballCounter > 3 && ballCounter < 5){
+          e_Feeder.setPosition(0);
+          pc_Feeder.setReference(0, ControlType.kPosition);
+        }
+      }
+      else if (newBallBoolean == true){
+        m_Intake.set(.25);
+      }
+      else if (ballDebounceBoolean == true){
+        ballDebounceBoolean = false;
+      }
+
+      else if (oldBallBoolean == true && newBallBoolean == false){
+        ballDebounceBoolean = true;
+      }
+
+      else{
+        m_Intake.set(1);
+        //j_XboxController.setRumble(RumbleType.kLeftRumble, 0);
+      }
+
+      if (ballCounter > 3){
+        Timer.delay(.5);
+        intakeExtended = false;
+        s_LeftIntake.set(false);
+        s_RightIntake.set(false);
+      }
+      else {
+        intakeExtended = true;
+        s_LeftIntake.set(true);
+        s_RightIntake.set(true);
+      }
+    }
+    
+    public void shootingBalls() {
+      ballCounter = 0;
+      m_BotShooter.setIdleMode(CANSparkMax.IdleMode.kCoast);
+      m_TopShooter.setIdleMode(CANSparkMax.IdleMode.kCoast);
+      if (e_BotShooter.getVelocity() > -5350){
+        m_BotShooter.set(-1);
+      }
+      else {
+        m_BotShooter.set(0);
+        readyToFeed = true;
+      }
+      if ( e_TopShooter.getVelocity() < 5350){
+        m_TopShooter.set(1);
+      }
+      else {
+        m_TopShooter.set(0);
+      }
+
+      if (readyToFeed = true){
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [QBA: Method assigns boolean literal in boolean expression (QBA_QUESTIONABLE_BOOLEAN_ASSIGNMENT)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#qba-method-assigns-boolean-literal-in-boolean-expression-qba-questionable-boolean-assignment).
This method assigns a literal boolean value (true or false) to a boolean variable inside an if or while expression. Most probably this was supposed to be a boolean comparison using ==, not an assignment using =.